datadog_checks_base/tests/test_openmetrics.py

```diff
@@ -1532,3 +1532,62 @@ def test_text_filter_input(mocked_prometheus_check, mocked_prometheus_scraper_co
 
     filtered = [x for x in check._text_filter_input(lines_in, mocked_prometheus_scraper_config)]
     assert filtered == expected_out
+
+
+@pytest.fixture()
+def mock_filter_get():
+    text_data = None
+    f_name = os.path.join(os.path.dirname(__file__), 'fixtures', 'prometheus', 'deprecated.txt')
+    with open(f_name, 'r') as f:
+        text_data = f.read()
+    with mock.patch(
+        'requests.get',
+        return_value=mock.MagicMock(
+            status_code=200,
+            iter_lines=lambda **kwargs: text_data.split("\n"),
+            headers={'Content-Type': text_content_type},
+        ),
+    ):
+        yield text_data
+
+
+class FilterOpentMetricsCheck(OpenMetricsBaseCheck):
+    def _filter_metric(self, metric):
+        return metric.documentation.startswith("(Deprecated)")
+
+
+@pytest.fixture
+def mocked_filter_openmetrics_check():
+    check = FilterOpentMetricsCheck('prometheus_check', {}, {})
+    check.log = logging.getLogger('datadog-prometheus.test')
+    check.log.debug = mock.MagicMock()
+    return check
+
+
+@pytest.fixture
+def mocked_filter_openmetrics_check_scraper_config(mocked_filter_openmetrics_check):
+    yield mocked_filter_openmetrics_check.get_scraper_config(PROMETHEUS_CHECK_INSTANCE)
+
+
+def test_filter_metrics(aggregator, mocked_filter_openmetrics_check,
+                        mocked_filter_openmetrics_check_scraper_config, mock_filter_get):
+    """ Tests label join GC on text format """
+    check = mocked_filter_openmetrics_check
+    mocked_filter_openmetrics_check_scraper_config['namespace'] = 'filter'
+    mocked_filter_openmetrics_check_scraper_config['metrics_mapper'] = {
+        'kube_pod_container_status_restarts': 'pod.restart',
+        'kube_pod_container_status_restarts_old': 'pod.restart'}
```

Comment:
```suggestion
        'kube_pod_container_status_restarts_old': 'pod.restart_old'}
```
Set the name to something else so that the `assert_all_metrics_covered()` will actually fail if this metric has not been filtered out correctly. The method will only check that every metric name has been asserted at least once.